api/deploy.go

```diff
@@ -400,3 +400,71 @@ func deployRebuild(w http.ResponseWriter, r *http.Request, t auth.Token) error {
 	}
 	return nil
 }
+
+// title: rollback update
+// path: /apps/{appname}/deploy/rollback/update
+// method: PUT
+// consume: application/x-www-form-urlencoded
+// responses:
+//   200: Rollback updated
+//   400: Invalid data
+//   403: Forbidden
+//   404: Not found
+//   409: Conflict
+func deployRollbackUpdate(w http.ResponseWriter, r *http.Request, t auth.Token) error {
+	appName := r.URL.Query().Get(":appname")
+	instance, err := app.GetByName(appName)
+	if err != nil {
+		return &tsuruErrors.HTTP{Code: http.StatusNotFound, Message: fmt.Sprintf("App %s was not found", appName)}
+	}
+	canUpdateRollback := permission.Check(t, permission.PermAppUpdateDeployRollback, contextsForApp(instance)...)
+	if !canUpdateRollback {
+		return &tsuruErrors.HTTP{
+			Code:    http.StatusForbidden,
+			Message: "User does not have permission to do this action in this app",
+		}
+	}
+	img := r.FormValue("image")
+	if img == "" {
+		return &tsuruErrors.HTTP{
+			Code:    http.StatusBadRequest,
+			Message: "you must specify an image",
+		}
+	}
+	rb := r.FormValue("enabled")
+	rollback, err := strconv.ParseBool(rb)
+	if err != nil {
+		return &tsuruErrors.HTTP{
+			Code:    http.StatusBadRequest,
+			Message: fmt.Sprintf("Status `enabled` set to: %s instead of `true` or `false`", rb),
+		}
+	}
+	reason := r.FormValue("reason")
+	if (reason == "") && (rollback == false) {
+		return &tsuruErrors.HTTP{
+			Code:    http.StatusBadRequest,
+			Message: "Reason cannot be empty while enabling rollback",
+		}
+	}
+	evt, err := event.New(&event.Opts{
+		Target:        appTarget(appName),
+		Kind:          permission.PermAppUpdateDeployRollback,
+		Owner:         t,
+		CustomData:    event.FormToCustomData(r.Form),
+		Allowed:       event.Allowed(permission.PermAppReadEvents, contextsForApp(instance)...),
+		AllowedCancel: event.Allowed(permission.PermAppUpdateEvents, contextsForApp(instance)...),
+		Cancelable:    false,
+	})
+	if err != nil {
+		return err
+	}
+	defer func() { evt.Done(err) }()
+	err = app.RollbackUpdate(instance, img, reason, rollback)
+	if err != nil {
+		return &tsuruErrors.HTTP{
+			Code:    http.StatusConflict,
```

Comment:
I dont think http.StatusConflict is the right choice here, e.g, if the image is not found we should probably return a BadRequest and for generic errors we should return InternalServerError.